documentation/smg/02_development_environment.md

```diff
@@ -1 +1,102 @@
+* Steps:
+  * From the Windows Start button navigate to Control Panel.
+  * In Control Panel under Category view, select Network and Internet, then Network and Sharing Center.
+  * Click the link Set up a new connection or network.
+  * In the new window, select Connect to a workplace, click Next
+  * Click on the selection Use my Internet connection (VPN)
+  * For Internet Address, use VPN.STFC.AC.UK. For destination name, use CICT VPN.
+  * If you see Next, click on it and fill in your federal ID and its password, with domain CLRC, click Finish.
+  * If you see Finish immediately, go to the Control Panel VPN page.  You will see CICT VPN near the top of the page. Click it to reveal Connect, Advanced options, and Remove button. Click Advanced Options, where you can fill in the ID, password and domain.
+  * Click Connect to establish the connection. This may need refreshing daily, or after shutdown or sleep, or just sometimes. Check back to the VPN page; if the Connect button is visible, you are not connected even if your email is working.
+
+#### Getting MATLAB
+
+MATLAB can be found at STFC RAL IT Support (Knowledge Base/MATLAB) at www.facilities.rl.ac.uk/support/KnowledgeBase/Matlab.aspx. This should set up an account with Matlab with your STFC email as id and a password of your choice. You can then go to the Matlab download page for STFC and download; choose R2018b or R2019b, which are the current versions supported at the time of writing. Or update to more recent if we have changed. This will give you an installation .exe.
+
+Running the .exe will download the files and install. You may be asked for activation details from Matlab. This will be your id and password as above. You are getting an academic individual STFC licence. Activation may also be required on first launch. Ideally you will be logged in as you with appropriate privileges. If not, you may need to run as administrator; not doing so puts you in an infinite loop of activate, launch, activate, launch.....
+
+This appears to be a one-user license and I am not necessarily expecting that you will be able to install on multiple machines e.g. personal and work machine. But it may work.
+
+NB I had problems trying to restart the download - it said I did not have access rights. Clearing cookies from the page solved this.
+
+#### Using MATLAB
+
+##### Prerequisites
+
+* You need to install `cmake`. I installed the current (at time of writing) version 3.18.2; there is a minimum version but all current versions should be OK for this. Select that `cmake` is on the path.
+* You need to install the Visual Studio SDK. This can be the actual SDK or a working VS installation. I installed Visual Studio 16 2019 community edition (it's free). As (currently) cloned from github, Horace will cache a reference to VS 2017 and will say that it cannot find VS. In explorer or otherwise, go to Horace/build in the Horace repo and delete or rename CMakeCache.txt before building. 
```

Comment:
>As (currently) cloned from github, Horace will cache a reference to VS 2017 and will say that it cannot find VS. In explorer or otherwise, go to Horace/build in the Horace repo and delete or rename CMakeCache.txt before building. 

Is this true? `CMakeCache.txt` is generated by CMake, so it should not exist until CMake has been run. Likewise with the `build` folder, which is created by the build script when it's run and does not exist in the repository

Reply:
CMakeCache.txt was present in the cloned repo with this issue. As noted, removing the file will cause it to be regenerated appropriately by CMake. Maybe we should remove the file from the repo. But it will be a problem that a new starter sees if they just start a build with a clone as it is at the moment.

Reply:
The file shouldn't be in Git -- and I can't see it on master in Herbert or Horace). 

This is clearly confusing so it may be worth making it clear that you *do* need to add the `-vs 2019` argument to `buid.ps1` calls, even though it's an optional argument to stop it defaulting to VS2017. If you change VS version it gets confused prompting that error and needing a clean/rebuild. It's possible you ran it once without while installing CMake leading to the cache being created.

Reply:
The 2017 default is included as that's the officially supported compiler and the one we're using on the build server at the moment

Reply:
Are you sure you didn't run the build script? That would create the `build` folder and `CMakeCache.txt`. The default for the build script is to attempt to build with Visual Studio 2017 - so that would explain everything. I'll open a ticket to not use 2017 as a default any more, not sure why we did that.

I can't find any `CMakeCache.txt` file in either Horace or Herbert repository.

Reply:
That must be right. The file is in the build folder - which isn't in a clone which hasn't been built. But I wasn't aware of running build prior to the problem, so looks as if current build will immediately generate this problem.